app/cluster/dynamic.go

```diff
@@ -17,15 +21,20 @@ var (
 )
 
 type ModeProvider interface {
```

Comment:
Now it is not only about `servermode`. What about
```
type EventChannelProvider interface {
	GetServerModeEventChannel(ctx context.Context) <-chan servermode.ChangeEvent
	GetWorkspaceIdsEventChannel(ctx context.Context) <-chan workspace.ChangeEvent
}
```

Reply:
Is not very idiomatic for Go to use the `Get` / `Set` prefix. The same goes for return type as a suffix `EventChannel`.

In general shorter names are preferred in Go.

Reply:
ignore get/set, only focus on the interface name :)

Reply:
what do you think of `StateChangeProvider`  ?